templates/htan-synapse-bucket.yaml

```diff
@@ -115,12 +120,15 @@ Resources:
               StringNotEqualsIgnoreCase:
                 s3:x-amz-grant-full-control: !Sub "id=${SynapseCanonicalId},id=${S3CanonicalUserId}"
               StringNotEquals:
-                s3:x-amz-grant-full-control: !Sub "id=${SynapseCanonicalId},id=${S3CanonicalUserId}"
+                s3:x-amz-grant-full-control: !Sub "id=${SynapseCanonicalId}, id=${S3CanonicalUserId}"
+              StringNotLike:
+                "aws:userid": !Join ["", [!Ref BucketServiceRoleArn, ":*"]]
```

Comment:
So the service role ARN can be used in place of the role ID?

Reply:
@clarisse-lau you're correct, we need the ID.

Reply:
good catch, fix in here and in depends on PR